applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nodal_displacement_response_function.h

```diff
@@ -193,8 +193,9 @@ class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) AdjointNodalDisplacementRespo
     ///@{
 
     std::string mTracedDofLabel;
-    PointTypePointer  mpTracedNode;
-    Element::Pointer mpNeighboringElement;
+    std::string mResponsePartName;
+    array_1d<double,3> mResponseDirection;
+    std::map<IndexType, std::vector<IndexType>> mElementNodeMap;
```

Comment:
```suggestion
    std::unordered_map<IndexType, std::vector<IndexType>> mElementNodeMap;
```

this is faster for index based access (which seems like you are doing)